core/src/main/java/org/infinispan/configuration/cache/TransactionConfiguration.java

```diff
@@ -186,6 +186,10 @@ public TransactionManagerLookup transactionManagerLookup() {
       return transactionManagerLookup.get();
    }
 
+   /**
+    * @deprecated since 9.0. it doesn't really works since it fetches the transaction manager when starting.
```

Comment:
Rewording: the transaction manager is only looked up once, during cache startup.